zebROS_ws/src/ros_control_boilerplate/src/frc_robot_interface.cpp

```diff
@@ -2570,6 +2585,139 @@ void FRCRobotInterface::write(const ros::Time& time, const ros::Duration& period
+				"candle->ConfigBrightnessScalar",
+				candle_state.getDeviceID()
+			)) {
+				ROS_INFO_STREAM("CANdle " << this->candle_names_[candle_id]
+						<< " : Set brightness to " << brightness);
+				candle_state.setBrightness(brightness);
+			} else {
+				candle_command.resetBrightnessChanged();
+			}
+		}
+
+		// Show status LED when active
+		//ROS_INFO_STREAM("Candle_status_2");
+		bool status_led_when_active;
+		if (candle_command.statusLEDWhenActiveChanged(status_led_when_active)) {
+			if (safeTalonCall(
+				candle->ConfigStatusLedState(!status_led_when_active),
+				"candle->ConfigStatusLedState",
+				candle_state.getDeviceID()
+			)) {
+				ROS_INFO_STREAM("CANdle " << this->candle_names_[candle_id]
+						<< " : Set status led when active to " << status_led_when_active);
+				candle_state.setStatusLEDWhenActive(status_led_when_active);
+			} else {
+				candle_command.resetStatusLEDWhenActiveChanged();
+			}
+		}
+
+		// Enabled
+		//ROS_INFO_STREAM("Candle_status_3");
+		bool enabled;
+		if (candle_command.enabledChanged(enabled)) {
+			if (safeTalonCall(
+				candle->configV5Enabled(enabled),
+				"candle->configV5Enabled",
+				candle_state.getDeviceID()
+			)) {
+				ROS_INFO_STREAM("CANdle " << this->candle_names_[candle_id]
+						<< " : Set enabled to " << enabled);
+				candle_state.setEnabled(enabled);
+			} else {
+				candle_command.resetEnabledChanged();
+			}
+		}
+
+		// Animation
+		//ROS_INFO_STREAM("Candle_status_4");
+		CANdleAnimation candle_animation;
+		Animation* animation;
+		if (candle_command.animationChanged(candle_animation)) {
+			ROS_INFO_STREAM("Candle_status_4.1");
+			// Convert from CANdleAnimation to the appropriate CTRE animation class
+			if (candle_animation.class_type == CANdleAnimationClass::BaseStandard) {
+				ROS_INFO_STREAM("Candle_status_4.1.1");
+				BaseStandardAnimation base_animation = candle_convert::convertBaseStandardAnimation(candle_animation);
+				ROS_INFO_STREAM("Candle_status_4.1.2");
+				animation = &base_animation;
+				ROS_INFO_STREAM("Candle_status_4.1.3");
+			} else {
+				ROS_INFO_STREAM("Candle_status_4.1.4");
+				BaseTwoSizeAnimation base_two_animation = candle_convert::convertBaseTwoAnimation(candle_animation);
+				ROS_INFO_STREAM("Candle_status_4.1.5");
+				animation = &base_two_animation;
+				ROS_INFO_STREAM("Candle_status_4.1.6");
+			}
+			ROS_INFO_STREAM("Candle_status_4.2");
+
+			if (safeTalonCall(
+				candle->Animate(*animation),
+				"candle->Animate",
+				candle_state.getDeviceID()
+			)) {
+				ROS_INFO_STREAM("CANdle " << this->candle_names_[candle_id]
+						<< " : Changed its animation" << animation->GetSpeed());
+				candle_state.setAnimation(candle_animation);
+			} else {
+				candle_command.resetAnimationChanged();
+			}
+			ROS_INFO_STREAM("Candle_status_4.done");
+		}
+
+		// LEDs
+		//ROS_INFO_STREAM("Candle_status_5");
+		vector<LEDGroup> led_groups;
+		if (candle_command.ledGroupChanged(led_groups)) {
```

Comment:
Think about the possibilities if both animations and led groups are used.  Should there be a call to disable animations if led groups are set?  Should there be an explicit clear / stop animation in the command interface?

Reply:
Likewise, should setting an animation first call setLED (0,0,0) on all the LEDs in range, or does this happen automatically?